lib/truffle/truffle/patching.rb

```diff
@@ -13,11 +17,16 @@ module Kernel
   alias_method :require_without_truffle_patching, :require
 
   def require(path)
+    if Truffle::Patching::TRUFFLE_BEFORE_PATCHES[path]
```

Comment:
> It does mean an extra Hash lookup for every require though.

Probably not a big deal given we scan the `$LOADED_FEATURES` Array everytime currently.